galley/pkg/runtime/resource/schema.go

```diff
@@ -0,0 +1,81 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package resource
+
+import (
+	"fmt"
+)
+
+// BaseTypeURL contains the common scheme & domain parts of all type urls.
+const BaseTypeURL = "types.googleapis.com"
+
+// Schema contains metadata about configuration resources.
+type Schema struct {
+	byKind map[Kind]Info
+	byURL  map[string]Info
+}
+
+// NewSchema returns a new instance of Schema.
+func NewSchema() *Schema {
+	return &Schema{
+		byKind: make(map[Kind]Info),
+		byURL:  make(map[string]Info),
+	}
+}
+
+// Register a proto into the schema.
+func (s *Schema) Register(protoName string) {
```

Comment:
s/protoName/messageName

Reply:
Made protoMessageName to be very explicit.